lib/bitreserve/base.rb

```diff
@@ -0,0 +1,15 @@
+module Bitreserve
+  class Base
+    def initialize(attributes = {})
+      instantiate_variables(attributes)
+    end
+
+    private
+
+    def instantiate_variables(attributes)
+      attributes.each_pair do |key, value|
+        instance_variable_set("@#{key}", value)
+      end
+    end
+  end
+end
```

Comment:
What separate directory? Maybe changing the name to `BaseModel` or `BaseEntity` would help.